pages/article/ArticleHandler.php

```diff
@@ -616,4 +625,39 @@ public function userCanViewGalley($request, $articleId, $galleyId = null)
         }
         return true;
     }
+
+    /**
+     * Format multilingual publication data for template:
+     * Default data at least includes in one language: full title, title, subtitle, keywords, abstract
+     * showMultilingualMetadataOpts adds multilingual metadata: title (by default includes fullTitle and subtitle), keywords, abstract, etc.
+     * showMetadataOpts: additional metadata
+     */
+    protected function getPublicationLocaleData(Publication $publication, string $contextPrimaryLocale, array $showMultilingualMetadataOpts, array $showMetadataOpts): array
+    {
+        $titles = collect([
+            'title' => $publication->getTitles('html'),
+            'subtitle' => $publication->getSubtitles('html'),
+            'fullTitle' => $publication->getFullTitles('html'),
+        ]);
+        $metadataOpts = collect($showMetadataOpts)->filter(fn ($opt) => in_array($opt, Services::get('schema')->getMultilingualProps(PKPSchemaService::SCHEMA_PUBLICATION)))
+            ->concat(['keywords', 'abstract'])->unique()->diff($titles->keys())->values();
+        $multilingualOpts = collect($showMultilingualMetadataOpts)
+            ->when(in_array('title', $showMultilingualMetadataOpts), fn ($m) => $m->concat(['subtitle', 'fullTitle'])->unique()->values());
+        $primaryLocale = isset($titles->get('title')[$contextPrimaryLocale]) ? $contextPrimaryLocale : $publication->getData('locale');
```

Comment:
Default view (no metadata options selected): the code tries to give the metadata based on the title language (if not ui then submission). If the metadata does not exist in title language, then the getLocalized data selects some found language. I tried to avoid to display a mix of languages as much as possible. I can change this to match the original logic, or I could display the other languages the same way like in the multilingual view.

Multilingual view (one or more options selected) : Using getLocalizedData won't work here because if in UI/submission language e.g. keywords do not exist, the function would show it in other language. In this view it would be inconsistent. 

I have not included author metadata in to this but I can do that